mythril/laser/ethereum/plugins/implementations/mutation_pruner.py

```diff
@@ -38,6 +39,10 @@ def initialize(self, symbolic_vm: LaserEVM):
         def sstore_mutator_hook(global_state: GlobalState):
             global_state.annotate(MutationAnnotation())
 
+        """FIXME: Check for changes in world_state.balances instead of adding MutationAnnotation for all calls.
```

Comment:
I don't think that works. 
Each transaction will by default modify the balances array. Just for unpayable ones the constraint `value == 0` get's added. Which then means the balances won't change.

So I think we can remove the fixme :sweat_smile:

Reply:
Part of this fix would be storing starting_balances *after* msg.value has
been added to the callee balance. Then we can ask the solver things like
"is it possible for balances to change in any way in the course of this
tx". If there's no value calls and the constraint [msg.value==0] on the end
state the answer would be "no" amirite?

Reply:
Ah, yeah misunderstanding on my part :sweat_smile: